gh-129847: Add graphlib.reverse(), graphlib.as_transitive() (#130875)
python/cpython: Doc/library/graphlib.rst

   .. versionadded:: 3.9

Graph Functions
---------------

Some functions are provided to work with directed acyclic graph structures using

> the same representation accepted by :meth:`TopologicalSorter.__init__`: a

Reviewer: Document creation fail here.

```suggestion
the same representation accepted by :meth:`!TopologicalSorter.__init__`: a
```

I'm not sure if that's right

Reviewer 2: Perhaps just better to say:

```suggestion
the same representation accepted by the :class:`TopologicalSorter` constructor: a
```

or 

```suggestion
the same representation accepted by :class:`TopologicalSorter`: a
```

Author: Huh, I don't see an error from `make -C doc html`.

Reviewer 2: We have a slightly odd setup to baseline reference warnings (see `check-warnings.py`), so CI runs once normally and then a second time with `-n` as an option to Sphinx.

mapping of nodes to iterables of predecessors.

Nodes can be any :term:`hashable` object. Duplicate edges are ignored.

For example, it can be useful to construct a TopologicalSorter that processes
a graph in reverse order. This can be done by passing the :func:`reverse` of a

> graph to the constructor of :class:`TopologicalSorter`::
> 
>    ts = TopologicalSorter(reverse(graph))

Reviewer 2: ```suggestion
graph to the constructor of :class:`TopologicalSorter`:

.. code-block:: python

   ts = TopologicalSorter(reverse(graph))
```

Author: Why? This is equivalent but 2 lines longer

Reviewer 2: We have a slight preference for the more explicit `code-block` directive in new documentation, though as you say both have the same result.

Author: Updated to use this throughout.
